cmd/podman/trust.go

```diff
@@ -132,7 +131,7 @@ func showTrustCmd(c *cli.Context) error {
 	if err := json.Unmarshal(policyContent, &policyContentStruct); err != nil {
 		return errors.Errorf("could not read trust policies")
 	}
-	policyJSON, err := trust.GetPolicyJSON(policyContentStruct, systemRegistriesDirPath)
+	policyJSON, showOutputMap, err := trust.GetPolicy(policyContentStruct, systemRegistriesDirPath)
```

Comment:
This is much better than the `repoval["type"].(string)` accesses before, but still the backend `trust` subpackage is asked to provide the formatting for both variants. It would be conceptually clearer if the backend subpackage provided the data (in well-typed structs), and this subpackage provided the formatting conversions (both JSON and non-JSON).